### PR TITLE
Listen to messages before iframe append

### PR DESCRIPTION
**Pull request by aghassemi**
Plus a few unrelated cleanups for video players.

### CODE REVIEW

File: extensions/amp-nexxtv-player/0.1/amp-nexxtv-player.js [hunk 1]
```diff
@@ -122,19 +115,19 @@ class AmpNexxtvPlayer extends AMP.BaseElement {
...
 
-    return this.loadPromise(this.iframe_)
-      .then(() => {
-        this.element.dispatchCustomEvent(VideoEvents.LOAD);
-        this.playerReadyResolver_(this.iframe_);
-      });
+    this.element.appendChild(this.iframe_);
+    this.playerReadyPromise_ = this.loadPromise(this.iframe_).then(() => {
+      this.element.dispatchCustomEvent(VideoEvents.LOAD);
+    });
```
Comment (jridgewell) [inline extensions/amp-nexxtv-player/0.1/amp-nexxtv-player.js]: 👏

File: extensions/amp-ooyala-player/0.1/amp-ooyala-player.js [hunk 2]
```diff
@@ -42,7 +42,7 @@ class AmpOoyalaPlayer extends AMP.BaseElement {
     this.playerReadyPromise_ = null;
 
     /** @private {?Function} */
-    this.unlistenMessage_ = null;
+    this.playerReadyResolver_ = null;
```
Comment (jridgewell) [inline extensions/amp-ooyala-player/0.1/amp-ooyala-player.js]: This can be dropped.

File: extensions/amp-youtube/0.1/amp-youtube.js [hunk 3]
```diff
@@ -191,12 +198,13 @@ class AmpYoutube extends AMP.BaseElement {
     );
 
     this.element.appendChild(this.iframe_);
-    this.playerReadyPromise_ = this.loadPromise(this.iframe_).then(() => {
```
Comment (jridgewell) [inline extensions/amp-youtube/0.1/amp-youtube.js]: Hm?

**Comment by aghassemi** [comment]
sorry for the spammy commits :) PTAL. I added back all the playerReadyPromise stuff, realized we need the playerReadyPromise in `buildCallback` so we can queue up actions calls until layout is done. (e.g. we want `<button on=tap:myVideo.mute>` to work even if `myVideo` has not gone through layoutCallback yet`)

**Comment by aghassemi** [comment]
oh man Travis is running for every one of my back-to-back commits. @erwinmombay @rsimha-amp Can Travis become smart to cancel previous in-progress runs when a new commit comes in?

**Comment by aghassemi** [comment]
@jridgewell Would a rejected promise drop references to all of its handlers? (wondering if ` load().then(...).catch( ()=> { this.playerReadyPromise = new Promise... } )` would be any better for memory footprint)

**Comment by aghassemi** [comment]
@jridgewell Also I think neither the current code nor your proposal will actually send the messages to iframe if load fails (resolving with a rejected promise still results in a rejected promise meaning all the queued up `sendCommands` won't run). But I think that's what we want anyway.

**Comment by jridgewell** [comment]
> Would a rejected promise drop references to all of its handlers? 

Yes, that's why I want you to resolve with a promise and not the promise's fulfilled value. Doing the good example above rejects the original `promise` if `otherPromise` rejects.

> wondering if load().then(...).catch( ()=> { this.playerReadyPromise = new Promise... } ) would be any better for memory footprint

This creates another promise instance, and takes an extra tick.

**Comment by jridgewell** [comment]
> Also I think neither the current code nor your proposal will actually send the messages to iframe if load fails

Yes. But the difference is the current code keeps all those messages in memory because the promise is never resolved (neither fulfilled or rejected) and could be fulfilled at some point in the future. Rejecting (by resolving with a rejected promise) will drop all those from memory.

**Comment by aghassemi** [comment]
Got it. PTAL.

File: extensions/amp-youtube/0.1/amp-youtube.js [hunk 4]
```diff
@@ -198,16 +197,14 @@ class AmpYoutube extends AMP.BaseElement {
...
 
-    this.win.addEventListener(
-      'message', event => this.handleYoutubeMessages_(event)
-    );
-
-    return this.loadPromise(iframe)
+    this.element.appendChild(this.iframe_);
+    const loadPromise = this.loadPromise(this.iframe_)
         .then(() => this.listenToFrame_())
         .then(() => {
```
Comment (jridgewell) [inline extensions/amp-youtube/0.1/amp-youtube.js]: Nit: Can we join this with the previous block?

**Comment by rsimha-amp** [comment]
@aghassemi, re: preventing Travis from running against every one of your commits, there seems to be hope: https://github.com/travis-ci/travis-ci/issues/964

I'll investigate this tomorrow and see if it's something we can institute within amphtml.

**Pull request merged**